ipa-core/src/protocol/basics/mul/semi_honest.rs

```diff
@@ -89,6 +91,26 @@ where
     Ok(Replicated::new_arr(lhs, rhs))
 }
 
+/// Implement secure multiplication for semi-honest contexts with replicated secret sharing.
+#[async_trait]
+impl<C, F, const N: usize> super::SecureMul<C> for Replicated<F, N>
```

Comment:
moved things around to avoid slapping descriptive-gate feature flag on too many things